app/projectsproxymodel.cpp

```diff
@@ -0,0 +1,101 @@
+/***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#include "projectsproxymodel.h"
+
+ProjectsProxyModel::ProjectsProxyModel( QObject *parent ) : QSortFilterProxyModel( parent )
+{
+}
+
+void ProjectsProxyModel::initialize()
+{
+  setSourceModel( mModel );
+  mModelType = mModel->modelType();
+
+  if ( mModelType == ProjectsModel::CreatedProjectsModel )
+    setFilterRole( ProjectsModel::ProjectName );
+  else
+    setFilterRole( ProjectsModel::ProjectFullName );
+
+  setFilterCaseSensitivity( Qt::CaseInsensitive );
+
+  sort( 0, Qt::AscendingOrder );
+}
+
+QString ProjectsProxyModel::searchExpression() const
+{
+  return mSearchExpression;
+}
+
+ProjectsModel *ProjectsProxyModel::projectSourceModel() const
+{
+  return mModel;
+}
+
+void ProjectsProxyModel::setSearchExpression( QString searchExpression )
+{
+  if ( mSearchExpression == searchExpression )
+    return;
+
+  mSearchExpression = searchExpression;
+  setFilterFixedString( mSearchExpression );
+  emit searchExpressionChanged( mSearchExpression );
+}
+
+void ProjectsProxyModel::setProjectSourceModel( ProjectsModel *sourceModel )
+{
+  if ( mModel == sourceModel )
+    return;
+
+  mModel = sourceModel;
```

Comment:
don't you need to call `setSourceModel` here?

Reply:
it is called on line 18 once the source model is initialized